posit-dev/positron
# Custom layout actions icons, notebook layout, + misc enhancements

### Intent

- Add icons for layout presets
- Move actions into the existing `"Customize Layout..."` quick-pick (See #3215)
<img width="631" alt="image" src="https://github.com/posit-dev/positron/assets/6764693/545bdf8d-81e3-4fe7-be5c-e7c6b1b164f6">

- Add lightweight notebook layout (see #3039)
<img width="1175" alt="image" src="https://github.com/posit-dev/positron/assets/6764693/42a42180-6760-4410-8a68-220424146688">

- Add the ability to clamp sizes around an ideal relative size. 
<img width="666" alt="image" src="https://github.com/posit-dev/positron/assets/6764693/3d38e178-e9bd-4fe9-8323-80083aedc623">

- Panel is now minimized instead of completely hidden from view. 
- Layouts have been tweaked a bit based on in-person discussions. 



### QA Notes

This PR removes the previous action for getting to the layouts. Now you can use the built-in `Customize Layout...` command or click the little layout icon in the upper right corner of Positron. 
Previous bugs where the side-by-side layout sometimes didn't actually hide the panel should be fixed as well.
